<?xml version="1.0"?>
<launch>

<arg name="unit_name" default="m3d_test"/>

<arg name="rot_laser_ip"   default="192.168.0.205"/>

<arg name="rot_laser_type"   default="LMS100C"/>




<group ns="$(arg unit_name)" clear_params="true">

	<arg name="vel"   default="12"/>

		<node name="rotation_laser" pkg="sick_minimal_driver" type="sick_minimal_driver" output="screen">
			<param name="ip" value="$(arg rot_laser_ip)"/>
			<param name="frame_id" value="$(arg unit_name)/rot_laser_optical"/>
			<param name="sa" value="-134"/>
			<param name="inv" value="false"/>
        		<remap from ="~laserScan" to="rot_scan"/>
		</node>

		<node name="rotation" pkg="m3dunit_base" type="m3d_driver_node" output="screen">
			<param name="encoderOffset" value ="0"/>
			<param name="m3d_frame_id" value="$(arg unit_name)/m3d_link"/>
			<param name="m3d_rot_frame_id" value="$(arg unit_name)/m3d_rot_laser_link"/>
			<param name="m3d_front_frame_id" value="$(arg unit_name)/m3d_front_laser_link"/>
        <param name="serial" value="/dev/ttyUSB0"/>
		</node>
<!--		<node name="aggregator" pkg="m3dunit_base" type="m3d_pc_aggregator" output="screen">-->
<!--			<remap to="/rotation_laser/laserScan" from="/aggregator/rotLaserScan"/>-->
<!--		</node>-->
    <node name="aggregator" pkg="m3d_aggregator" type="m3d_aggregator" output="screen">
        <param name="pointCloudFrame" value="$(arg unit_name)/m3d_link"/>
        <param name="rotLaserPointCloud" value="rot_scan"/>
		</node>
		<node name="output_topic1" pkg="rostopic" type="rostopic" output="screen" args="pub /$(arg unit_name)/rotation/velocity std_msgs/Float32 $(arg vel)"/>

		<node name="tf_broadcaster" pkg="m3dunit_base" type="transformBroadcaster.py" output="screen">
			<param name="front_laser_type" value="none"/>
			<param name="rot_laser_type"   value="$(arg rot_laser_type)"/>
			<param name="prefix" value="$(arg unit_name)/"/>
		</node>

</group>
</launch>
